src/content/docs/ru/tutorial/5-astro-api/2.mdx

```diff
@@ -0,0 +1,322 @@
+
+    ```astro title="src/pages/tags/[tag].astro" ins={3-5}
+    <BaseLayout pageTitle={tag}>
+      <p>Посты с тегом {tag}</p>
+      <ul>
+        {filteredPosts.map((post) => <li><a href={post.url}>{post.frontmatter.title}</a></li>)}
+      </ul>
+    </BaseLayout>
+    ```
+
+4. Вы даже можете рефакторить этот код, чтобы использовать ваш компонент `<BlogPost />` вместо этого! (Не забудьте импортировать этот компонент в начале файла `[tag].astro`.)
+
+    ```astro title="src/pages/tags/[tag].astro" del={4} ins={5}
+    <BaseLayout pageTitle={tag}>
+      <p>Посты с тегом {tag}</p>
+      <ul>
+        {filteredPosts.map((post) => <li><a href={post.url}>{post.frontmatter.title}</a></li>)}
+        {filteredPosts.map((post) => <BlogPost url={post.url} title={post.frontmatter.title}/>)}
+      </ul>
+    </BaseLayout>
+    ```
+
+5. Проверьте предварительный просмотр в вашем браузере для ваших отдельных страниц с тегами, и теперь вы должны увидеть список всех ваших блог-постов, содержащих этот конкретный тег.
+
+<Box icon="question-mark">
+
+### Анализируйте шаблон
+
+Для каждого из следующих укажите, написан ли код **внутри** функции `getStaticPath()`, или **вне** её.
+
+1. Вызов `Astro.glob()` для получения информации обо всех ваших файлах `.md`, которую нужно передать каждому маршруту страницы.
+
+    <MultipleChoice>
+    <Option isCorrect>внутри `getStaticPaths`</Option>
+    <Option>вне `getStaticPaths`</Option>
+    </MultipleChoice>
+
+2. Список маршрутов, которые должны быть сгенерированы (возвращены) `getStaticPaths()`
+
+    <MultipleChoice>
+    <Option isCorrect>внутри `getStaticPaths`</Option>
+    <Option>вне `getStaticPaths`</Option>
+    </MultipleChoice>
+
+3. Полученные значения `props` и `params`, которые будут использоваться в HTML-шаблоне.
+
+    <MultipleChoice>
+    <Option>внутри `getStaticPaths`</Option>
+    <Option isCorrect>вне `getStaticPaths`</Option>
+    </MultipleChoice>
+</Box>
+
+:::note[Вывод]
+Если вам нужна информация для создания маршрутов страниц, напишите её **внутри** `getStaticPaths()`.
+
+Чтобы получить информацию в HTML-шаблоне маршрута страницы, напишите её **вне** `getStaticPaths()`.
+:::
+
+
+## Продвинутый JavaScript: Генерация страниц из существующих тегов
+
+Теперь ваши страницы с тегами статически определены в `[tag].astro`. Если вы добавите новый тег к сообщению в блоге, вам также придется вернуться на эту страницу и обновить маршруты страниц.
+
+Следующий пример показывает, как заменить код на этой странице кодом, который будет автоматически искать и генерировать страницы для каждого тега, используемого на ваших страницах блога.
+
+:::note
+Даже если это выглядит сложно, вы можете попробовать следовать инструкциям и самостоятельно построить эту функцию! Если вы не хотите сейчас разбираться с необходимым JavaScript, вы можете перейти к [готовому примеру кода](#final-code-sample) и использовать его напрямую в своем проекте, заменяя существующий контент.
```

Comment:
Since the heading is translated, it changes the slug, which we can update here.
```suggestion
Даже если это выглядит сложно, вы можете попробовать следовать инструкциям и самостоятельно построить эту функцию! Если вы не хотите сейчас разбираться с необходимым JavaScript, вы можете перейти к [готовому примеру кода](#пример-окончательного-кода) и использовать его напрямую в своем проекте, заменяя существующий контент.
```